src/components/Highlights.vue

```diff
@@ -1,26 +1,181 @@
 <template>
   <div class="highlights-bar">
-    <div class="betas">JSE Index:{{ jseIndex }}</div>
-    <div class="betas">{{ industryIndices[1] }}: {{ industryBetas[1] }}</div>
-    <div class="betas">{{ industryIndices[2] }}: {{ industryBetas[2] }}</div>
+    <div class="betas">
+      All Share Index (30 Days): {{ index1 }}
+      <span
+        id="allShareNumber"
+        v-bind:class="{
+          positive: index1Change >= 0,
+          negative: !(index1Change >= 0),
+        }"
+        v-on:mount="index1Sign = index1Change <= 0"
+      >
+        ({{ index1Change }}%)
+      </span>
+    </div>
+
+    <div class="betas">
+      TOP 40 Index (30 Days): {{ index2 }}
+      <span
+        id="allShareNumber"
+        v-bind:class="{
+          positive: index2Change >= 0,
+          negative: !(index2Change >= 0),
+        }"
+        v-on:mount="index1Sign = index2Change <= 0"
+      >
+        ({{ index2Change }}%)
+      </span>
+    </div>
+
+    <div class="betas">
+      F&I Index (30 Days): {{ index3 }}
+      <span
+        id="allShareNumber"
+        v-bind:class="{
+          positive: index3Change >= 0,
+          negative: !(index3Change >= 0),
+        }"
+        v-on:mount="index3Sign = index3Change <= 0"
+      >
+        ({{ index3Change }}%)
+      </span>
+    </div>
+
+    <div class="betas">
+      Industrials Index (30 Days): {{ index4 }}
+      <span
+        id="allShareNumber"
+        v-bind:class="{
+          positive: index4Change >= 0,
+          negative: !(index4Change >= 0),
+        }"
+        v-on:mount="index4Sign = index4Change <= 0"
+      >
+        ({{ index4Change }}%)
+      </span>
+    </div>
+
+    <div class="betas">
+      Resources Index (30 Days): {{ index5 }}
+      <span
+        id="allShareNumber"
+        v-bind:class="{
+          positive: index5Change >= 0,
+          negative: !(index5Change >= 0),
+        }"
+        v-on:mount="index5Sign = index5Change <= 0"
+      >
+        ({{ index5Change }}%)
+      </span>
+    </div>
   </div>
 </template>
 
 <script>
+//Axios stuff
+import axios from 'axios'
+
+var period = 22
+
 export default {
   name: 'Highlights',
   data() {
     return {
-      industryBetas: ['0.7276 (5%)', '0.9 (-7%)', '1.1 (25%)', '0.68 (-30%)'],
-      industryIndices: [
-        'Oil and Gas',
-        'Chemicals',
-        'Mining',
-        'Forestry and Paper',
-      ],
-      jseIndex: '+15%',
+      period: 22,
+
+      index1: [1, 2, 3],
+      index1Change: [1, 2, 3],
+      index1Sign: false,
+
+      index2: [1, 2, 3],
+      index2Change: [1, 2, 3],
+      index2Sign: false,
+
+      index3: [1, 2, 3],
+      index3Change: [1, 2, 3],
+      index3Sign: false,
+
+      index4: [1, 2, 3],
+      index4Change: [1, 2, 3],
+      index4Sign: false,
+
+      index5: [1, 2, 3],
+      index5Change: [1, 2, 3],
+      index5Sign: false,
     }
   },
+  mounted() {
+    //Getting the info for JSE All Share Index
+    axios
```

Comment:
I would make a general method for each of these so that it makes it more readable. You can create a general function that takes in the share as an input.

E.g. (sorry my syntax may be wrong been doing a lot of PHP coding lately lol)
getIndexInfo( indexCode) {
  axios.get(`/api/shares?share={indexCode}`).then(...)
}

Then you could use this function for all of them and it would be clear from the index code what index it is getting so you wouldn't need comments either.

Reply:
Sure, I gave that a shot but I was doing something wrong somewhere so I decided to give it a skip for the sake of time but happy to go back and do this